src/comparator/wrappers.py

```diff
@@ -181,22 +181,29 @@ def label(self):
         )
         # FieldDescriptorProto.label has field number 4.
         return WithLocation(
-            FieldDescriptorProto().Label.Name(self.field_pb.label),
+            label_repeated,
             self.source_code_locations,
             self.path + (4,) if label_repeated else self.path,
         )
 
     @property
-    def required(self) -> bool:
+    def required(self):
         """Return True if this field is required, False otherwise.
 
         Returns:
             bool: Whether this field is required in field_behavior annotation.
         """
-        return (
+        required = (
             field_behavior_pb2.FieldBehavior.Value("REQUIRED")
             in self.field_pb.options.Extensions[field_behavior_pb2.field_behavior]
         )
+        # fmt: off
+        return WithLocation(
+            required,
+            self.source_code_locations,
+            self.path + (8, 1052),
```

Comment:
I think it might be more readable in some future PRs to define a dictionary which maps the options to the numbers, like `{"service": 3, "oauth_scope": 1050, ...}`, then you can use the option names instead of using the number directly.

Reply:
SGTM! That could be a refactor once the location path are all added so we can summarize them to the map.